### PR TITLE
Feature/activity log types

### PR DESCRIPTION
**Pull request by sorja**
## Description
Create types/interfaces for ActivityLog
Add test for ActivityLogFactory

Minor: Refactor some imports on entry `index.ts`

## Types of changes

- [ ] Docs change
- [x] Refactoring
- [ ] Dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How has this been tested
Unit tests

#### Do you consider this PR needs further testing?

- [x] No
- [ ] Yes

## Disclaimer

Part of #CU-j76qcq

### CODE REVIEW

**Comment by minotogna** [comment]
Task linked: [CU-j76qcq Survey create](https://app.clickup.com/t/j76qcq)

File: src/activityLog/factory.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  id: number
+  userUuid: string
+  type: ActivityLogType
+  content: any
+  system: boolean
+}
+
+export const ActivityLogFactory: Factory<ActivityLog<ActivityLogType, any>, ActivityLogFactoryParams> = {
+  createInstance: (params: ActivityLogFactoryParams): ActivityLog<ActivityLogType, any> => {
+    const defaultProps = {
```
Comment (SteRiccio) [inline src/activityLog/factory.ts:14]: it should be defaultParams

**Review by SteRiccio — APPROVED** [review]

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/activityLog/factory.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export type ActivityLogFactoryParams = {
+  id: number
+  userUuid: string
+  type: ActivityLogType
+  content: any
+  system: boolean
+}
+
+export const ActivityLogFactory: Factory<ActivityLog<ActivityLogType, any>, ActivityLogFactoryParams> = {
```
[the remark this answers is not in the document]
Comment (sorja) [reply src/activityLog/factory.ts:12]: Let's keep factory (as discussed in daily)

File: src/activityLog/activityLog.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  // User
+  userInvite = 'userInvite',
+  userUpdate = 'userUpdate',
+  userRemove = 'userRemove',
+
+  // Analysis
+  chainCreate = 'chainCreate',
+  chainPropUpdate = 'chainPropUpdate',
+  chainNodeDef = 'chainNodeDef',
```
Comment (minotogna) [inline src/activityLog/activityLog.ts:54]: please, see in arena

File: src/activityLog/factory.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { Factory } from '../common'
+import { ActivityLog, ActivityLogType } from './activityLog'
+
+export type ActivityLogFactoryParams = {
+  id: number
+  userUuid: string
+  type: ActivityLogType
+  content: any
+  system: boolean
```
Comment (minotogna) [inline src/activityLog/factory.ts:9]: system is optional

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by SteRiccio — DISMISSED** [review]

File: src/activityLog/factory.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+export type ActivityLogFactoryParams = {
+  id: number
+  userUuid: string
+  type: ActivityLogType
+  content: any
+  system?: boolean
+}
+
+export const ActivityLogFactory: Factory<ActivityLog<ActivityLogType, any>, ActivityLogFactoryParams> = {
+  createInstance: (params: ActivityLogFactoryParams): ActivityLog<ActivityLogType, any> => {
```
Comment (minotogna) [inline src/activityLog/factory.ts:13]: you need to find a way to use generics for type and content.

File: src/survey/index.ts [hunk 6]
```diff
@@ -3,9 +3,12 @@ export type { Survey, SurveyProps, SurveyCycle, SurveyDependencyGraph, SurveyDep
...
 export { SurveyFactory } from './factory'
+export type { SurveyFactoryParams } from './factory'
 
 export type { SurveyService } from './service'
 
 export type { SurveyRefData } from './refData'
 
 export { SurveyRefDataFactory } from './refData'
+
+export { validateNewSurvey } from './surveys/validator'
```
Comment (minotogna) [inline src/survey/index.ts:14]: you need to export the SurveyValidator object. not each single method

File: src/survey/service.ts [hunk 7]
```diff
@@ -22,6 +22,8 @@ export interface SurveyService {
 
   getMany(options: { limit?: number; offset?: number; user: User }): Promise<Array<Survey>>
 
+  getManyByName(options: { surveyName: string }): Promise<Array<Survey> | null>
```
Comment (minotogna) [inline src/survey/service.ts:25]: name should be added to getMany method. do not add another method

File: src/survey/surveys/index.ts [hunk 8]
```diff
@@ -1,10 +1,12 @@
...
 import { getCategoryItemByCodePaths } from './refsData'
+import { validateNewSurvey } from './validator'
 
 export const Surveys = {
   getNodeDefByName,
   getNodeDefByUuid,
   getNodeDefParent,
   isNodeDefAncestor,
   getCategoryItemByCodePaths,
+  validateNewSurvey,
```
Comment (minotogna) [inline src/survey/surveys/index.ts:11]: create a SurveyValidator object.

**Review by minotogna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**